pandas/io/stata.py

```diff
@@ -2060,16 +2091,17 @@ class StataWriter(StataParser):
 
     def __init__(
         self,
-        fname,
-        data,
-        convert_dates=None,
-        write_index=True,
-        byteorder=None,
-        time_stamp=None,
-        data_label=None,
-        variable_labels=None,
-    ):
+        fname: FilePathOrBuffer,
+        data: DataFrame,
+        convert_dates: Optional[Dict[Hashable, str]] = None,
```

Comment:
can you use Label from pandas._typing

Reply:
I looked at Label.  Is Optional[Hashable] the same? Isn't Optional[Hashable] just None or some hashable object?

Reply:
the problem is that None is hashable but not recognised as such by mypy. we therefore need Union[Hashable, None] to allow us to pass None where Hashable is allowed. In pandas we don't use Union[..., None] for stylistic reasons, so this become Optional[Hashable]

Reply:
> Isn't Optional[Hashable] just None or some hashable object?

yes, but these are valid for index labels and names

Reply:
I see the difference now, thanks.  I've updated Hashable->Label.